### DISCUSSION TITLE
Is a leaderboard a good usecase for Incremental Static Regeneration?

### DISCUSSION THREAD

**Discussion by florianwalther-private** [vercel/next.js#38755] [Help] [answered]
I want to display a leaderboard on my website but it would be wasteful to make a new database request whenever a user opens that page.
I was planning to use ISR here and refresh the leaderboard only every 30-60 seconds.

Do you think that's a good alternative to caching the data server-side? Or are there any problems with this approach?

**Comment by icyJoseph** [comment, marked as answer]
Sounds good to me. You could even trigger revalidation upon certain events, like a new leader so that it is shown ASAP.

**Reply by florianwalther-private** [reply]
Yea I've played around with this idea too! Thank you for your answer!

**Comment by GabenGar** [comment]
Do you need SEO for this leaderboard? If not, then don't bother with server render and do it on the client.

**Reply by florianwalther-private** [reply]
I want to avoid making a database call whenever a user opens this page. In other words, I want to cache it in some way. Caching it via getStaticProps seems easier than caching it on the server.

**Reply by GabenGar** [reply]
ISR is still jerking server and the database. Client-render will allow you to use websockets for things like this which is infinitely better than whatever html-rendering strategy you can come up with.
